src/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -272,8 +272,12 @@ public static Version Version
         }
 
 #if !FEATURE_PAL
-        private static Lazy<bool> s_IsWindows8OrAbove = new Lazy<bool>(() => 
+        private static bool? s_IsWindows8OrAbove;
```

Comment:
Assignment of `bool?` is not atomic. This has a race condition bug.

Reply:
The common ways to fix this are:

1. Tri-state integer or enum, e.g. https://github.com/dotnet/coreclr/blob/e5ff28022e99b391ca75867c1a378beee0e594f2/src/System.Private.CoreLib/shared/System/Globalization/TextInfo.cs#L34
2. Wrap the static field in a helper type with a single readonly field, e.g.  https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/src/System/ApplicationModel.Windows.cs#L14. The advantage is that the tiered JIT can tell that the value is readonly and optimize out the check completely. I think this one would be best fit here. (It is ok for the helper type to be nested in Environment.)
3. `StrongBox<bool>`, e.g. https://github.com/dotnet/corefx/blob/master/src/System.Console/src/System/Console.cs#L135. The C# compiler caches the delegate lambda in a helper type as well, so this is just a more expensive version of `(2)`. It may be nice to fix Console to (1) or (2) instead.
4. `Lazy<bool>` even more expensive than (3).

Reply:
Gone for 2; just verifying it skips the .cctor also

Reply:
Works a treat, thank you :)